File: 2025/d1/p1.py
with open("test.txt") as f:
    instr = []
    for raw in f:
        s = raw.strip()
        direction = s[0]
        steps = int(s[1:])
        instr.append([direction, steps])

init = 50
res = 0
for i in range(len(instr)):
    direction, steps = instr[i]
    
    if direction == "R":
        init = (init + steps) % 100
        if init == 0:
            res += 1
    elif direction == "L":
        init = (init - steps) % 100
        if init == 0:
            res += 1

print(res)

File: 2025/d1/p1.cpp
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

int main(){

    ifstream file("test.txt");

    int init = 50;
    int res = 0;

    string line;
    while (getline(file, line)){
        char dir = line[0];
        int steps = stoi(line.substr(1));

        if (dir == 'R'){
            init = (init + steps) % 100;
            if (init == 0) res ++;
        }
        else if (dir == 'L'){
            init = (init - steps) % 100;
            if (init == 0) res ++;
        }

    }
    file.close();   
    cout << res << endl;

    return 0;
}